src/csm_testing/tests/iuf_apis/__main__.py

```diff
@@ -0,0 +1,799 @@
+        print("~"*50)
+        print("INFO: Total test cases passed: {} test cases skipped: {}".format(count,total_count-count))
+        print("~"*50)
+        sys.exit(1)
+
+    print(f"INFO: Get history/time for activity: {activity} , time:{time}")
+    try:
+        history_time = apis.get_activity_history_time(activity,time) 
+    except Exception as ex:
+        print(f"ERROR: Unable to get history/time for activity: {activity} , time:{time}")
+        print(f"ERROR: {ex}")
+        print("~"*50)
+        print("INFO: Total test cases passed: {} test cases skipped: {}".format(count,total_count-count))
+        print("~"*50)
+        sys.exit(1)
+
+    if history_time is not None:
+        print(history_time)
+        count += 1
+    else :
+        print("ERROR: history/time for activity: {activity} , time:{time} not found")
+        print("~"*50)
+        print("INFO: Total test cases passed: {} test cases skipped: {}".format(count,total_count-count))
+        print("~"*50)
+        sys.exit(1)
+
+def get_activity_session(apis, activity):
+    global count , total_count
+    print("TEST CASE: Get Activity Session")
+    if not apis.activity_exists(activity):
+        print(f"ERROR: Activity {activity} does not exist.")
+        print("~"*50)
+        print("INFO: Total test cases passed: {} test cases skipped: {}".format(count,total_count-count))
+        print("~"*50)
+        sys.exit(1)
+    sessions = apis.get_activity_sessions(activity).json()
+    if sessions is not None:
+        session = sessions[0]["name"]
+    else :
+        print(f"ERROR: Sessions not found for activity: {activity}")
+        print("~"*50)
+        print("INFO: Total test cases passed: {} test cases skipped: {}".format(count,total_count-count))
+        print("~"*50)
+        sys.exit(1)
+
+    print(f"INFO: Get activity/session for activity: {activity} , session:{session}")
+
+    try:
+        session = apis.get_activity_session(activity,session)
+        print(session)
+        count += 1 
+    except Exception as ex:
+        print(f"ERROR: Unable to get activity/session for activity: {activity} , session:{session}")
+        print(f"ERROR: {ex}")
+        print("~"*50)
+        print("INFO: Total test cases passed: {} test cases skipped: {}".format(count,total_count-count))
+        print("~"*50)
+        sys.exit(1)
+
+def main():
+    try:
+        os.makedirs(MEDIA_DIR, exist_ok=True)
+        print(f"INFO: Directory {MEDIA_DIR} created successfully.")
+    except OSError as e:
+        print(f"ERROR: Failed creating directory {MEDIA_DIR}: {e}")
+        sys.exit(1)
+    
+    try:
+        shutil.copy(f"{tar_dir}/{FOLDER_NAME}.tar.gz", MEDIA_DIR)
+        shutil.copy(f"{tar_dir}/product_vars.yaml", MEDIA_DIR)
+        shutil.copy(f"{tar_dir}/management-bootprep.yaml", MEDIA_DIR)
+        print(f"INFO: Files copied successfully to {MEDIA_DIR}.")
+    except IOError as e:
+        print(f"ERROR: Failed copying files: {e}")
+        sys.exit(1)
```

Comment:
This code is almost identical to lines 45-60 in `iuf_run/__main__.py`. Unless there is a good reason not to, common code should be moved into shared modules under `src/csm_testing/lib`, and then imported by the tests.